Enhance check-in and check-out blocks with meta and post types

tibiii opened the pull request:
---
pr: 645
related_issues:
  - 644
blocks:
  - lsx/checkin-time
  - lsx/checkout-time
post_types:
  - accommodation
author: tibiii
labels:
  - enhancement
  - block
  - audit
---

## Summary
This pull request updates both the Check In Time and Check Out Time block variations for the LSX Tour Operator plugin, making them more descriptive, context-aware, and visually consistent. The changes ensure the blocks only register for the `accommodation` post type, add richer metadata and examples, and switch to a custom icon component for improved branding.

Block registration improvements:

* Both `checkin-time` and `checkout-time` block variations now register only when editing an `accommodation` post type, using a polling mechanism to ensure correct timing. [[1]](diffhunk://#diff-03e2ffa42a0f603db3109211fb112aef77b5173aa7eea62192f9496ab7630bc0L2-R70) [[2]](diffhunk://#diff-ca0d438bfc80f02cef0b501b2e8e7d9916bff8336a05da5676932a9044d02565L2-R70)
* The icon for both blocks is changed from a static image to the custom `lsx-tour-operator/icons` block with `checkInAccommodationIcon`, improving consistency and maintainability. [[1]](diffhunk://#diff-03e2ffa42a0f603db3109211fb112aef77b5173aa7eea62192f9496ab7630bc0L2-R70) [[2]](diffhunk://#diff-ca0d438bfc80f02cef0b501b2e8e7d9916bff8336a05da5676932a9044d02565L2-R70)

Block configuration enhancements:

* Added `description`, `icon`, `keywords`, and a detailed `example` to both `block.json` files, improving discoverability and usability in the block editor. [[1]](diffhunk://#diff-b5e2f161b97730fb892688482970de4b43157c89fa75930ba00b14a833f1f346R4-R43) [[2]](diffhunk://#diff-9422c80c827d810a9c2a2e6c56591c01714d0e44977e178e5b6cf1f6b19a1900R4-R43)
* The block examples now show a heading and sample time, providing a clearer preview in the editor. [[1]](diffhunk://#diff-b5e2f161b97730fb892688482970de4b43157c89fa75930ba00b14a833f1f346R4-R43) [[2]](diffhunk://#diff-9422c80c827d810a9c2a2e6c56591c01714d0e44977e178e5b6cf1f6b19a1900R4-R43)

Technical and metadata updates:

* Added a `textdomain` field to both block configurations for proper localization support. [[1]](diffhunk://#diff-b5e2f161b97730fb892688482970de4b43157c89fa75930ba00b14a833f1f346R4-R43) [[2]](diffhunk://#diff-9422c80c827d810a9c2a2e6c56591c01714d0e44977e178e5b6cf1f6b19a1900R4-R43)
* Updated asset version hashes in the PHP asset files to reflect the new builds. [[1]](diffhunk://#diff-9dc533a8859923daf6892f6e3d95b2ab22b2634de43681536bd42c2343ce4f4fL1-R1) [[2]](diffhunk://#diff-33d3fa10bf17104f02baac4d6bc91e361478a62ce3a98852dc1d4eb890c9e78aL1-R1)…d improved registration logic
## Scope
- [x] PHP
- [x] JS/Blocks
- [ ] Styles (theme.json / selectors)
- [ ] Tests (Playwright)
- [ ] Docs

## Screenshots / Videos
_If relevant._

## Checklist
- [x] Block API v3
- [x] `block.json` updated (title, description, keywords, supports, example, selectors)
- [ ] Registered via PHP using metadata
- [ ] Theme.json styles (no bespoke CSS unless unavoidable)
- [x] Preview works in inserter
- [ ] E2E tests added/updated

## Related issues
This closes #644 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enhanced Check‑in and Check‑out blocks with icons, translated labels, nested grouped layouts and example previews.

* **Improvements**
  * Blocks now register only in Accommodation contexts (including relevant templates) with retry/timing and isActive gating for reliable editor loading.
  * Added internationalisation, descriptions and keywords for better discoverability.
  * Build assets declare wp‑i18n dependency.

* **Documentation**
  * Example content added to illustrate block usage.

* **Chores**
  * Internal global storage variable renamed for content model paths.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tibiii committed:
Enhance check-in and check-out blocks with descriptions, examples, and improved registration logic

tibiii committed:
Update check-in and check-out time content to reflect correct times

tibiii committed:
Fix check-in time block variation description and update version number

krugazul committed:
Fixing the incorrect content Model path for the vendor.

krugazul committed:
Adding in the phpcs ignore.

tibiii committed:
Improve registration logic to show in accommodation related templates as well

tibiii committed:
Merge branch 'refactor/checkin-checkout-blocks' of github.com:lightspeedwp/tour-operator into refactor/checkin-checkout-blocks

tibiii committed:
Localise all strings

krugazul reviewed (approved):
@tibiii Ive reviewed and tested the code changes,  Im happy for this to be merged in.

tibiii committed:
Update changelog with recent feature additions

Added icon block icons for checkin/checkout times and a filter for relevant post types. Updated changelog with recent changes and integrations.

Signed-off-by: Tibi Buzdugan <buzdugan.tiberiu@gmail.com>
